File: InPlan.Web/Modules/Tenencia/Areas/AreasRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Tenencia;

[ConnectionKey("Default"), Module("Tenencia"), TableName("Areas")]
[DisplayName("Áreas Trabajo"), InstanceName("Área")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed partial class AreasRow : Row<AreasRow.RowFields>, IIdRow, INameRow
{
    const string jIdEmpresa = nameof(jIdEmpresa);

    class RowTemplate
    {
        [DisplayName("Id Area"), Identity, IdProperty]
        public int? IdArea { get; set; }

        [LookupEditor(typeof(EmpresasRow))]
        [DisplayName("Empresa"), NotNull, ForeignKey("Empresas", "IdEmpresa"), LeftJoin(jIdEmpresa), TextualField(nameof(IdEmpresaDescripcionCorta))]
        public int? IdEmpresa { get; set; }

        [DisplayName("Descripción"), Size(200), NotNull, QuickSearch, NameProperty]
        public string Descripcion { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }

        [DisplayName("Empresa "), Expression($"{jIdEmpresa}.[DescripcionCorta]")]
        public string IdEmpresaDescripcionCorta { get; set; }
    }
}

File: InPlan.Web/Modules/ServerTypes/Tenencia/AreasRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface AreasRow {
    IdArea?: number;
    IdEmpresa?: number;
    Descripcion?: string;
    Observaciones?: string;
    Baja?: boolean;
    IdEmpresaDescripcionCorta?: string;
}

export abstract class AreasRow {
    static readonly idProperty = 'IdArea';
    static readonly nameProperty = 'Descripcion';
    static readonly localTextPrefix = 'Tenencia.Areas';
    static readonly lookupKey = 'Tenencia.Areas';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<AreasRow>('Tenencia.Areas') }
    static async getLookupAsync() { return getLookupAsync<AreasRow>('Tenencia.Areas') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<AreasRow>();
}